lib/Sema/ConstraintSystem.h

```diff
@@ -2728,6 +2748,48 @@ class DisjunctionChoice {
     return decl->isOperator() ? decl : nullptr;
   }
 };
+
+class ConstraintBucket {
```

Comment:
I'm very glad to have this functionality split out into a separate class, but please document what this class does via comments.